File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, n = nums.length - 1;
        
        while(l <= n) {
            int m = l + ((n - l)/2);
            if (target < nums[m]) {
                n = m - 1;
            } else if (target > nums[m]) {
                l = m + 1;
            } else {
                return m;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_i = 0

        for i in range(len(nums)):
            if max_i < i:
                return False
            curr_i = nums[i] + i
            max_i = max(max_i, curr_i)
        
        return max_i >= len(nums) - 1